### PR TITLE
Adding support for sending request headers in azure core rest request API

### PR DESCRIPTION
**Pull request by aasimkhan30**
This feature is added to support sending additional headers in all azure rest requests. It will be used by migration extension to send the session id in all the requests we are making for SQL migration resources.

### CODE REVIEW

File: extensions/azurecore/src/azureResource/utils.ts [hunk 1]
```diff
@@ -333,8 +333,9 @@ export async function getSelectedSubscriptions(appContext: AppContext, account?:
  * @param requestBody Optional request body to be used in PUT and POST requests.
  * @param ignoreErrors When this flag is set the method will not throw any runtime or service errors and will return the errors in errors array.
  * @param host Use this to override the host. The default host is https://management.azure.com
+ * @param requestHeaders Provide additional request headers
```
Comment (Charles-Gagnon) [inline extensions/azurecore/src/azureResource/utils.ts:336]: Update to `customHeaders` or make customHeaders be requestHeaders instead (regardless, make sure all the params are the same for the other places this is used too)

File: extensions/azurecore/src/azureResource/utils.ts [hunk 2]
```diff
@@ -375,11 +376,14 @@ export async function makeHttpRequest(account: azdata.Account, subscription: azu
 		return result;
 	}
 
+	const requestHeaders = {
+		'Content-Type': 'application/json',
+		'Authorization': `Bearer ${securityToken.token}`,
+		...customHeaders
```
Comment (Charles-Gagnon) [inline extensions/azurecore/src/azureResource/utils.ts:382]: Is this handling undefined customHeaders properly?

Comment (aasimkhan30) [reply extensions/azurecore/src/azureResource/utils.ts:382]: Yes. Nothing gets added if the customHeaders is undefined.

Comment (Charles-Gagnon) [reply extensions/azurecore/src/azureResource/utils.ts]: Ah, guess that happens when there's other properties in the object. Just doing `...undefined` by itself throws an error.

I think it'd be better to still give this a default of an empty object though

`customHeaders: ... = { }`

File: extensions/azurecore/src/extension.ts [hunk 3]
```diff
@@ -236,8 +236,9 @@ export async function activate(context: vscode.ExtensionContext): Promise<azurec
 			requestType: azurecore.HttpRequestMethod,
 			requestBody: any,
 			ignoreErrors: boolean,
-			host: string = 'https://management.azure.com'): Promise<azurecore.AzureRestResponse> {
-			return azureResourceUtils.makeHttpRequest(account, subscription, path, requestType, requestBody, ignoreErrors, host);
+			host: string = 'https://management.azure.com',
+			requestHeaders: { [key: string]: string } = {}): Promise<azurecore.AzureRestResponse> {
```
Comment (brian-harris) [inline extensions/azurecore/src/extension.ts:240]: Does ADS support different clouds or only public cloud?  if so this api URL would need to be based on current cloud.

**Review by brian-harris — APPROVED** [review]

File: extensions/azurecore/src/extension.ts (hunk 3 above)
Comment (Charles-Gagnon) [reply extensions/azurecore/src/extension.ts:240]: That's a good catch @brian-harris! Yeah @aasimkhan30 this code should be using the currently configured auth settings to get the correct proper ARM endpoint.

**Review by Charles-Gagnon — APPROVED** [review]

**Review by kburtram — APPROVED** [review]

**Pull request merged**